eva-accession-release-automation/gather_clustering_counts/gather_release_counts.py

```diff
@@ -0,0 +1,300 @@
+import argparse
+import csv
+import glob
+import os
+from collections import defaultdict, Counter
+from functools import lru_cache
+
+from ebi_eva_common_pyutils.command_utils import run_command_with_output
+from ebi_eva_common_pyutils.common_utils import pretty_print
+from ebi_eva_common_pyutils.logger import logging_config, AppLogger
+from ebi_eva_common_pyutils.metadata_utils import get_metadata_connection_handle
+from ebi_eva_common_pyutils.pg_utils import get_all_results_for_query
+
+
+logger = logging_config.get_logger(__name__)
+
+
+shell_script_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'bash')
+
+assembly_table_name = 'eva_stats.release_rs_statistics_per_assembly'
+
+def find_link(key_set, dict1, dict2, source_linked_set1=None, source_linked_set2=None):
+    """
+    Assuming 2 dictionaries providing respectively the list of values linked to a key, and a list of keys linked to a
+    value, (in our case this is a list of assemblies linked to a taxonomy and the list of taxonomy linked to a assembly)
+    , this recursive function starts from one of the value and find all the related keys and values and provided them
+    in 2 frozen sets. For any key that belong to a relationship, this should provide the same pair of frozensets
+    regardless of the starting key.
```

Comment:
**Optional:** A small example could be helpful here in understanding this function..

Reply:
Agreed. You could maybe even include this example in the form of a test case...